CryptoBinPkg/Driver/readme.md

```diff
@@ -1,54 +1,99 @@
+
+   You need to select what flavor of crypto works for your platform for each phase. Cross reference the PCD flavor
+   files and your platform's requirements to find which one works best. The `ALL` flavor enables all supported
+   cryptographic functions.
+
+   If you require a crypto algorithm not available in any flavor, including `ALL`, file a feature request issue in
+   this repo describing the algorithm required and a use case for that algorithm in platform firmware.
+
+2. **Your Platform** Architecture
+
+    For each phase, ensure you have selected the correct architecture. Failure to do so will potentially result in
+    build errors and incorrect behaior during boot.
+
+3. Dependencies Built into Shared Crypto
+
+    Ultimately, the shared crypto binaries have dependencies that must be fulfilled by the shared crypto project
+    when the binaries are built. It is important to be aware of those selections and how they may impact usage of
+    the binary.
+
+    **Feedback** on dependency selections is welcome.
 
-2. What ARCH to use and what they're compiled with
-    Currently all the different ARCHs use VS2022 for compilation with AARCH64 being the notable exception.  This may
-    have platform remifications so be mindful of that.
-    NOTE: AARCH64 will be compiled with VS2022 eventually and GCC5 binaries will no longer be produced.
+    There are a couple of specific services (library selections) that you need to be aware of when it comes to crypto
+    interaction.
 
-3. Library consideration
-    There are a couple of specific libraries that you need to be aware of when it comes to crypto interaction
-    - DebugLib is set in the binary itself and is not effected by the platform.  Currently the MM crypto is
-      using a null debug lib
-    - RngLib does effect the crypto binaries.  Which RngLib you use will have effects on crypto.  For example for
-      AARCH64 platforms ArmTrngLib and BaseRngLib effect eachother which causes issues.  Dependencies like that can
-      lead to complications while integrating the crypto binaries.
+    - **Debug Output** - The binary for each boot phase is linked against a `DebugLib` instance.
+      - Currently the `Null` instance of `DebugLib` is used for most phases which means that those binaries do not emit
+        debug output.
+      - The `PEIM` binaries are currently linked against the [`PeiDxeDebugLibReportStatusCode](https://github.com/tianocore/edk2/tree/master/MdeModulePkg/Library/PeiDxeDebugLibReportStatusCode)
+      library instance which sends debug output to the report status code infrastructure.
+
+      - The `DXE_DRIVER` binaries are currently linked against the [`UefiDebugLibDebugPortProtocol`](https://github.com/tianocore/edk2/tree/master/MdePkg/Library/UefiDebugLibDebugPortProtocol)
+      library instance which sends messages to an instance of the
+      [`EFI_DEBUGPORT_PROTOCOL`](https://github.com/tianocore/edk2/blob/master/MdePkg/Include/Protocol/DebugPort.h).

```

Comment:
Should these links be against mu_basecore instead of edk2? These binaries are relevant to both, but I would assume most consumers are going to be general mu consumers

Reply:
Updated to mu_basecore.